src/librustc/middle/reachable.rs

```diff
@@ -47,6 +48,17 @@ fn generics_require_inlining(generics: &ty::Generics) -> bool {
     false
 }
 
+struct HasAnonTy(bool);
+
+impl<'tcx> TypeVisitor<'tcx> for HasAnonTy {
+    fn visit_ty(&mut self, t: ty::Ty<'tcx>) -> bool {
+        if let ty::TyAnon(..) = t.sty {
+            self.0 = true;
+        }
+        t.super_visit_with(self)
```

Comment:
Please use the flags machinery in `rustc::ty` instead. There might already be a flag and method for it there.